Networking/Networking/Model/Product/Product.swift

```diff
@@ -463,7 +463,11 @@ public struct Product: Codable, GeneratedCopiable, Equatable {
         try container.encode(sku, forKey: .sku)
         try container.encode(manageStock, forKey: .manageStock)
         try container.encode(soldIndividually, forKey: .soldIndividually)
-        try container.encode(stockQuantity, forKey: .stockQuantity)
+
+        // API currently only accepts integer values for stock quantity
+        let integerStockQuantity = Int(truncating: stockQuantity as NSDecimalNumber? ?? 0)
```

Comment:
[Small Tip] you can use `map`, to convert an optional type without having to deal with coalescing operators, which makes the code a bit hard to read.
```swift
let integerStockQuantity = stockQuantity.map { NSDecimalNumber(decimal: $0).intValue }
```

Reply:
Thanks, that's a great tip! I wasn't familiar with this as a way to avoid dealing with coalescing operators.

I'll leave this as-is for now (in this PR), since https://github.com/woocommerce/woocommerce-ios/pull/3717 changes how we do this encoding.